### PR TITLE
Add Agile & Strong Scroll held items for Hisui

### PR DESCRIPTION
**Pull request by CypherX**
## Description
Adds the Hisui held items as described in #4845. Current modifier values are placeholders. Definitely needs testing to find a balance.

~~Tweaked how click attack is calculated to support modifying the contribution of a single pokemon. Benchmarking showed that this new method is faster than the original calculation.~~
Moved into separate PR

## How Has This Been Tested?
Equipped each held item and checked that the pokemon attack and click attack increased or decreased accordingly

## Screenshots (optional):
image
image

## Types of changes
- New Held Items

### CODE REVIEW

File: src/scripts/party/PartyPokemon.ts [hunk 1]
```diff
@@ -117,6 +118,12 @@ class PartyPokemon implements Saveable {
                 this.addOrRemoveHeldItem(this.heldItem());
             }
         });
+        this.clickAttackBonus = ko.computed((): number => {
+            // Caught + Shiny + Resistant
+            const bonus = 1 + +this.shiny + +(this.pokerus >= GameConstants.Pokerus.Resistant);
+            const heldItemMultiplier = this.heldItem && this.heldItem() instanceof HybridAttackBonusHeldItem ? (this.heldItem() as HybridAttackBonusHeldItem).clickAttackBonus : 1;
```
Comment (RegisCoaxans) [inline src/scripts/party/PartyPokemon.ts:124]:
```suggestion
            const heldItemMultiplier = this.heldItem() instanceof HybridAttackBonusHeldItem ? (this.heldItem() as HybridAttackBonusHeldItem).clickAttackBonus : 1;
```
First check is truish at all time due to `this.heldItem` pointing to the observable. However, it means it is not required as the second check sorts out `undefined` anyway.

Comment (CypherX) [reply src/scripts/party/PartyPokemon.ts:124]: I copied it from the regular pokemon attack held item bonus check, lol. Thanks

File: src/scripts/party/PartyPokemon.ts [hunk 2]
```diff
@@ -117,6 +118,12 @@ class PartyPokemon implements Saveable {
                 this.addOrRemoveHeldItem(this.heldItem());
             }
         });
+        this.clickAttackBonus = ko.computed((): number => {
+            // Caught + Shiny + Resistant
+            const bonus = 1 + +this.shiny + +(this.pokerus >= GameConstants.Pokerus.Resistant);
```
[the remark this answers is not in the document]
Comment (CypherX) [reply src/scripts/party/PartyPokemon.ts:123]: Out of the 4 methods I benchmarked pureComputed was the slowest, including slowing than the original calculation, while computed was the quickest.

**Review by Qwertypop04 — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**